bdb-ip.md

```diff
@@ -0,0 +1,1000 @@
+When a BigchainDB user is querying BigchainDB for either spent or unspent outputs with specific public keys
+And some specific public keys do not match any existing either spent or unspent outputs in BigchainDB
+Then BigchainDB returns for all either spent or unspent outputs matching specific public keys all required fractals of information to generate a follow-up TRANSFER transaction
+And the BigchainDB user is not notified about potentially not matched either spent or unspent outputs.
+
+Note: See clarification in Scenario 2.
+
+
+#### Use case 3: Sha256Threshold and Ed25519 outputs mixed
+
+As a BigchainDB user
+I want to retrieve all Sha256Threshold outputs and/or Ed25519 outputs from multiple transactions for given public keys
+so that I can create a follow-up TRANSFER transaction.
+
+
+##### Scenario 3.1: Query for single or multiple non-existing Ed25519 or Sha256Treshold output(s) with single or multiple public keys
+
+Given non-existent transactions
+When a BigchainDB user is querying BigchainDB for spent or unspent outputs with a single or multiple public keys
+And no output in any transactions in BigchainDB matches the given public key(s)
+Then a BigchainDB user is notified that no outputs for given public key(s) exist.
+
+
+##### Scenario 3.2: Query for single or multiple either spent or unspent Ed25519 or Sha256Threshold outputs with single or multiple keys
+
+Given a valid transaction with single or multiple either spent or unspent Sha256Threshold outputs composed from one or more Ed25519 outputs or single or multiple either spent or unspent Ed25519 outputs exists
+When a BigchainDB user is querying BigchainDB for spent or unspent output(s) with specific public key(s)
+Then BigchainDB returns all required fractals of information to generate a follow-up TRANSFER transaction for the existing matching output(s).
+
+
+#### Use case 4: Result atomicity and verifiability
+
+There is two competing arguments:
+
+- [/outputs should return atomic and verifiable objects](https://github.com/bigchaindb/bigchaindb/issues/835)
+- [/outputs should return 'usable', optional verifiable objects](https://github.com/bigchaindb/bigchaindb/issues/1227)
+
+A potential solution could introduce an `safe=true|false` query parameter
+(where default `safe=true` to include a list of outputs into the result (that
+are potentially unsafe to use as a node could lie to a user). Querying a
+transaction from 2/3 of the network and comparing hashes is the only safe
+option currently.
+
+
+#### Use case 5: (Un)Spendable flag for specific public key
+
+As a BigchainDB user
+I want to retrieve outputs based on public keys I flag as either spent or unspent.
+
+@TimDaub: "It's not clear if this is a confirmed use case."
+
+
+#### Use case 6: [Outputs filtered by specific asset(s)](https://github.com/bigchaindb/bigchaindb/issues/1473)
+
+- Add `?asset_id` to query parameters and filter by
+
+
+#### Proposed /outputs endpoint design
```

Comment:
@jmduque: 

We're currently having an ongoing design process to update the /outputs endpoint to something more useful. I tried to model the endpoint heavily to the proposal you made on gitter a while ago. I'd love to hear your feedback on it.